spec/models/export_rule_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'rails_helper'
+
+RSpec.describe ExportRule, type: :model do
+  describe '#new' do
+    context 'when parameters are correct' do
+      subject { FactoryBot.create(:export_rule) }
+
+      it { is_expected.to be_a ExportRule }
+
+      its(:dynamic_field_group)  { is_expected.to be_a DynamicFieldGroup }
+      its(:field_export_profile) { is_expected.to be_a FieldExportProfile }
+      its(:translation_logic) do
+        is_expected.to be_json_eql %(
+          [
+            {
+              "render_if": { "present": ["role"] },
+              "element": "mods:name",
+              "content": "{{role}}"
+            }
+          ]
+        )
+      end
+    end
+
+    context 'when missing dynamic_field_group' do
+      let(:export_rule) { FactoryBot.build(:export_rule, dynamic_field_group: nil) }
+
+      it 'does not save' do
+        expect(export_rule.save).to be false
+      end
+
+      it 'returns correct error' do
+        export_rule.save
+        expect(export_rule.errors.full_messages).to include 'Dynamic field group is required'
+      end
+    end
+
+    context 'when missing field_export_profile' do
+      let(:export_rule) { FactoryBot.build(:export_rule, field_export_profile: nil) }
+
+      it 'does not save' do
+        expect(export_rule.save).to be false
+      end
+
+      it 'returns correct error' do
+        export_rule.save
+        expect(export_rule.errors.full_messages).to include 'Field export profile is required'
+      end
+    end
+
+    context 'when missing translation_logic' do
+      let(:export_rule) { FactoryBot.create(:export_rule, translation_logic: nil) }
+
+      it 'adds empty translation_logic' do
+        expect(export_rule.translation_logic).to be_json_eql "[]"
+      end
+    end
+
+    context 'when translation_logic is invalid' do
+      let(:export_rule) { FactoryBot.build(:field_export_profile, translation_logic: 'randomstring') }
+
+      it 'does not save' do
+        expect(export_rule.save).to be false
+      end
+
+      it 'returns correct error' do
+        export_rule.save
+        expect(export_rule.errors.full_messages).to include 'Translation logic does not validate as JSON'
+      end
+    end
+
+    context 'when creating duplicate export rule' do
+      let(:export_rule_1) { FactoryBot.create(:export_rule) }
+      let(:export_rule_2) do
+        FactoryBot.build(
+          :export_rule,
+          dynamic_field_group: export_rule_1.dynamic_field_group,
+          field_export_profile: export_rule_1.field_export_profile
+        )
+      end
+
+      it 'does not save' do
+        expect(export_rule_2.save).to be false
+      end
+
+      it 'returns correct error' do
+        export_rule_2.save
+        expect(export_rule_2.errors.full_messages).to include 'Dynamic field group has already been taken'
```

Comment:
I think this is a case where an overridden error message make sense, since this doesn't fully describe the error to the user.  Maybe something like this? "Dynamic field group already has an export rule for field_export_profile #{field_export_profile.name}" ?

Reply:
Sure. I was thinking about it, but could not think of a good error message. I'll update!